lib/declare_schema/model/index_definition.rb

```diff
@@ -83,9 +83,11 @@ def each_hash(result)
             end
           end
 
-          pk_index = connection.indexes(table).find { |index| index.name.to_s == PRIMARY_KEY_NAME } or return nil
-
-          Array(pk_index.columns)
+          if (pk_index = connection.indexes(table).find { |index| index.name.to_s == PRIMARY_KEY_NAME })
+            Array(pk_index.columns)
+          elsif model.connection.columns(table).any? { |col| col.name == 'id' }
+            ['id']
```

Comment:
This is the right answer 99.999% of the time, so we might as well fall back to it.

Reply:
Can we ask the model what the `_declared_primary_key` or use Rails' built in methodology for looking up with the primary key for a given model is here? I agree that 99.999% of the time this is correct, but I also don't think we need to go fishing for the primary key index in the database do we?

Reply:
Yes, we need to look in the database because that's the _only_ reason this method is there. We have `_declared_primary_key` for when we want to know what was declared.

Want to change any method names to help make this clear?

BTW I'm looking forward to deleting some of this code as soon as we stop supporting Rails 4. This method will live on, but only for `sqlite`.

Reply:
No, don't think any renaming is necessary. What you explained makes sense, a common mixup that's easy to happen throughout this code is differentiating between the cases where you need to access the declared vs the actual. Thank you for the explanation, I don't think any changes are necessary here and look forward to the day of this not being as complicated.